### ISSUE TITLE
Expand documentation of RegionData object graph and ComposedSort paths

### ISSUE THREAD

**Issue by albertbori** [WeakAuras/WeakAuras2#3076]
**Is your feature request related to a problem? Please describe.**
I'm attempting to learn to use the custom sorting functions (including composition-sort) in dynamic groups. I've read the documentation here: https://github.com/WeakAuras/WeakAuras2/wiki/Custom-Dynamic-Group#regiondata

The problem is that I'm unable to see all possible composition paths available to sort by. The object graph sample gives "hand-waivy" example values, like `region = {...}`. Further down in the documentation, it gives the helpful composition-style examples of the following, but fails to enumerate all possible paths:

```
WeakAuras.ComposeSorts(
  WeakAuras.SortAscending{"region", "state", "stacks"},
  WeakAuras.SortDescending{"region", "state", "expirationTime"}
)
```

**Describe the solution you'd like**
I would love to have more complete documentation of the possible paths to use for the composed-sort functions. (or at least a more complete version, within reason)

**Describe alternatives you've considered**
I spent a few hours trying to print out these objects on the fly to inspect them, but it is very difficult to do from within the game. I even wrote my own custom lua table dump, but the information was unintelligible to the layman.

**Comment by emptyrivers** [comment]
Yeah, the regionData objects are not exactly "user-friendly" - this is somewhat by design. The "region" and "data" fields of those objects are living things that WeakAuras uses in many places throughout the addon, and I don't especially wish for users to feel comfortable poking around in there - that's a hood way for ✨weird things✨ to appear, which become complicated to maintain around.

The dynamic group functions expose "raw" internal machinery like this for two reasons:

- builtin functionality is constructed to use the same machinery, and adding extra bells to safe-ify custom code is (so far) not worth the trouble
- I am curious to see what neat things people might come up with. Useful (but missing) features sometimes reveal themselves from the ways in which clever users "break" WeakAuras.

For a proper understanding of what those objects are & what information they contain, consider examining the source code [here](https://github.com/WeakAuras/WeakAuras2/blob/main/WeakAuras/RegionTypes/DynamicGroup.lua) which creates & manages the regionData objects.
